### DISCUSSION TITLE
Smithy models in scala code string

### DISCUSSION THREAD

**Discussion by ShahOdin** [disneystreaming/smithy4s#1480] [General]
I want to report what I think is a bug/confusing behaviour in smithy4s. What's the recommended way of submitting a minimal example. Could it be done via cli with the smithy model coded in scala code as string? I know we can pass a plugin to scala cli, so maybe that works if there is a way of generating smithy code from string? 🤔 🤷🏽

**Comment by kubukoz** [comment]
I'm not sure I understand the question, but if you're just asking how to provide a reproduction for an issue you're seeing, I would recommend a GH repo with an sbt build. I create branches for reproductions all the time in https://github.com/kubukoz/demos, for example.

**Reply by kubukoz** [reply]
About scala-cli, it only supports compiler plugins right now, and we need something that runs beforehand - and scala-cli doesn't support calling code generators automatically, yet. There's an issue for that: https://github.com/VirtusLab/scala-cli/issues/610
